Coralogix/Sources/Model/InstrumentationData.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  InstrumentationData.swift
+//
+//
+//  Created by Coralogix Dev Team on 29/07/2024.
+//
+
+import Foundation
+
+struct InstrumentationData {
+    let otelSpan: OtelSpan
+    let otelResource: OtelResource
+ 
+    init(otel: SpanDataProtocol, labels: [String: Any]?) {
+        self.otelSpan = OtelSpan(otel: otel, labels: labels)
+        self.otelResource = OtelResource(otel: otel)
+    }
+    
+    func getDictionary() -> [String: Any] {
+        var result = [String: Any]()
+        result[Keys.otelSpan.rawValue] = self.otelSpan.getDictionary()
+        result[Keys.otelResource.rawValue] = self.otelResource.getDictionary()
+        return result
+    }
+}
+
+struct OtelSpan {
+    let spanId: String
+    let traceId: String
+    let name: String
+    let attributes: [String: Any]
+    let startTime: [UInt64]
+    let endTime: [UInt64]
+    let status: [String: Any]
+    let kind: Int
+    let duration: [UInt64]
+    
+    init(otel: SpanDataProtocol, labels: [String: Any]?) {
+        self.spanId = otel.getSpanId() ?? ""
+        self.traceId = otel.getTraceId() ?? ""
+        self.name = otel.getName() ?? ""
+        var attributes = [String: Any]()
+        
+        if let otelAttributes = otel.getAttributes() {
+            attributes = attributes.merging(otelAttributes) { (_, new) in new }
+        }
+        
+        if let labels = labels {
+            attributes = attributes.merging(labels) { (_, new) in new }
+        }
+        self.attributes = attributes
+        let currentDate = Date().timeIntervalSince1970
+        let defualtTime = [UInt64(currentDate), 0]
+        if let startTime = otel.getStartTime() {
+            let (integerPart, fractionalPart) = modf(startTime)
+            self.startTime = [UInt64(integerPart), UInt64(fractionalPart * 1_000_000_000)]
+        } else {
+            self.startTime = defualtTime
+        }
+        
+        if let endTime = otel.getEndTime() {
+            let (integerPart, fractionalPart) = modf(endTime)
+            self.endTime =  [UInt64(integerPart), UInt64(fractionalPart * 1_000_000_000)]
+        } else {
+            self.endTime = defualtTime
+        }
+        
+        self.status = otel.getStatusCode()
+        self.kind = otel.getKind()
+        self.duration = [0, 0]
```

Comment:
even that we saw the backend kind of ignore the value, it might change in the future
lets pass a valid duration